### PR TITLE
Users can join bets

### PR DESCRIPTION
**Pull request by ghost**
Pivotal Links: 
https://www.pivotaltracker.com/story/show/140512067
https://www.pivotaltracker.com/story/show/141507823

Dependent on #35  
Each BetOption now has a "Bet!" button, which takes a user to the page where they can enter their betting amount and place it or cancel and return to the Bet page.

### CODE REVIEW

File: app/views/bets/show.html.erb
Comment (ghost) [inline app/views/bets/show.html.erb]: This section has been deleted in one of the previous PR's already, so it should get resolved upon merging.

File: app/views/shared/_bet_display.html.erb
Comment (qr8r) [inline app/views/shared/_bet_display.html.erb]: Nice!

File: app/controllers/user_bets_controller.rb
Comment (qr8r) [inline app/controllers/user_bets_controller.rb]: `@user_bet.find_or_initialize_by(stuff)`

File: app/views/user_bets/new.html.erb
Comment (qr8r) [inline app/views/user_bets/new.html.erb]: Checking for a logged in user should happen at the controller level.
Could we also make this work if the option doesnt exist

File: app/views/user_bets/new.html.erb
Comment (qr8r) [inline app/views/user_bets/new.html.erb]: Could be a select box, or radio buttons with the specified option selected

File: app/views/user_bets/new.html.erb
Comment (qr8r) [inline app/views/user_bets/new.html.erb]: Using the find_or_initialize above will make it so you dont have to specify the values for all of these fields

File: app/controllers/user_bets_controller.rb
Comment (qr8r) [inline app/controllers/user_bets_controller.rb]: Find or initialize again

**Review by qr8r — CHANGES_REQUESTED** [review]

**Comment by qr8r** [comment]
@YarekTenko Its hard to review this pr since the creator relationship is not defined on Bet.

I will pulll it down and review once that gets merged in

**Comment by qr8r** [comment]
Jenkins test this please

**Comment by cbrunsdon** [comment]
jenkins test this please

**Comment by jacquesporveau** [comment]
This branch is out of date with the base branch. Please update.

**Comment by qr8r** [comment]
@YarekTenko  still seeing undefined method `creator' for #<Bet:0x007fb185949768> when attempting to test this

**Comment by ghost** [comment]
@brendandeere Thanks, Brendan, it should be fixed now.

**Comment by qr8r** [comment]
https://www.pivotaltracker.com/story/show/140512067/comments/167080481 

I would like to see this implemented prior to merging ^^

**Comment by jacquesporveau** [comment]
This branch is out of date with master. Please update.

**Comment by qr8r** [comment]
placing a user bet with no amount selected gives `nil can't be coerced into BigDecimal`
placing a bet with no option selected gives the cryptic message `Couldn't place your bet`

File: app/views/bets/show.html.erb
Comment (qr8r) [inline app/views/bets/show.html.erb]: Method should go on the form element
:-1: inline styles. Styles should always be written in external style sheets

File: config/routes.rb
Comment (qr8r) [inline config/routes.rb]: You can change this line so that you dont need to include so many empty views

**Comment by ghost** [comment]
Wasn't verification a separate story?

**Comment by qr8r** [comment]
@YarekTenko Possibly, we have a few stories that specify validation. The big thing for me is if you submit a user be with no amount it breaks the bet/show page. 

If you want to resolve the empty text issue by creating a separate story for that, Im on board. But I wont merge a PR that actually breaks a page

**Comment by ghost** [comment]
@brendandeere Ah, gotcha, thanks, Brendan! Hope I fixed it this time.

**Comment by qr8r** [comment]
Submitting an empty form leads to a 500 `Render and/or redirect were called multiple times in this action.` 

Please Try and break your own code prior to making a pull request. Test the interactions, enter good values, enter incomplete values, enter invalid values, enter no values, click buttons multiple times, etc

**Comment by YarekTenko** [comment]
Woops, accidentally merged #66 into it, sorry, I thought it was based on master.

File: app/views/bets/show.html.erb
Comment (jacquesporveau) [inline app/views/bets/show.html.erb]: I would like to see a *hint* in here to show how my entry should be formatted. I tried to bet "one hundred dollars" and I'll I got was a flash telling me to enter a valid amount. If you had a placeholder in this text field to show them the format you are expected I think the user would have a better experience.

File: app/views/bet_options/_bet_option_radio.html.erb
Comment (jacquesporveau) [inline app/views/bet_options/_bet_option_radio.html.erb]: What is this closing `td` tag corresponding to?

File: app/views/shared/_bet_display.html.erb
Comment (jacquesporveau) [inline app/views/shared/_bet_display.html.erb]: Misusing a header tag. Should not be inside the p.

Comment (ghost) [reply app/views/shared/_bet_display.html.erb]: Thanks! Hopefully fixed it.

**Review by qr8r — APPROVED** [review]

**Pull request merged**
